src/google-classroom-extractor/google_classroom_extractor/mapping/user_submission_activities.py

```diff
@@ -0,0 +1,154 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import json
+from typing import Dict
+from pandas import DataFrame, concat, Series
+from google_classroom_extractor.mapping.constants import (
+    SOURCE_SYSTEM,
+    ENTITY_STATUS_ACTIVE,
+)
+
+
+def submissions_to_user_submission_activities_dfs(
+    submissions_df: DataFrame,
+) -> Dict[str, DataFrame]:
+    """
+    Convert a Submission API DataFrame to a Dict of UserActivity
+    UDM DataFrames grouped by source system section id.
+
+    Parameters
+    ----------
+    submissions_df: DataFrame
+        is a Submission API DataFrame
+
+    Returns
+    -------
+    Dict[str, DataFrame] LMS UDM UserActivity DataFrames
+        grouped by source system section id
+
+    Notes
+    -----
+    UserActivity DataFrame columns are:
+        ActivityDateTime: The date/time the activity occurred
+        ActivityStatus: The activity status
+        ActivityTimeInMinutes: The total activity time in minutes
+        ActivityType: The type of activity, here "Submission" or "Grade"
+        AssignmentIdentifier: A unique numeric identifier assigned to the assignment
+        Content: Content associated with the activity
+        EntityStatus: The status of the record
+        LMSSectionIdentifier: A unique numeric identifier assigned to the section
+        SourceSystem: The system code or name providing the user data
+        SourceSystemIdentifier: A unique number or alphanumeric code assigned to a
+            user by the source system
+        LMSUserIdentifier: A unique numeric identifier assigned to the user
+        LMSUserActivityIdentifier: A unique numeric identifier assigned to the user activity
+        CreateDate: Date this record was created
+        LastModifiedDate: Date this record was last updated
+    """
+    assert isinstance(submissions_df, DataFrame)
+    assert "submissionHistory" in submissions_df.columns
+    assert "id" in submissions_df.columns
+    assert "courseId" in submissions_df.columns
+    assert "courseWorkId" in submissions_df.columns
+
+    # convert json-like submissionHistory string to list of dicts
+    submissions_df["submissionHistory"] = submissions_df["submissionHistory"].apply(lambda json_like: json.loads(json_like.replace("'", '"')))
```

Comment:
@lalrg I see that Brad is also doing a swap of `'` and `"` for consistency.

Reply:
Yeah, json.loads() complains otherwise. Doing exec() was another option, but no way am I going to execute something pulled from an API 😱